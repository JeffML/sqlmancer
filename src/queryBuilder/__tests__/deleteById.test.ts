import { withDialects } from './__utilties__'
import { ActorDeleteByIdBuilder } from './__fixtures__/models'

describe('DeleteBuilder', () => {
  withDialects((options, rollback) => {
    describe('basic queries', () => {
      test('no additional options', async () => {
        const builder = new ActorDeleteByIdBuilder(options, 20)
        const { sql, bindings } = builder.toQueryBuilder().toSQL()
        await rollback(builder, result => expect(result).toBeBoolean())
        expect(sql).toMatchSnapshot()
        expect(bindings).toMatchSnapshot()
      })
    })
  })
})
