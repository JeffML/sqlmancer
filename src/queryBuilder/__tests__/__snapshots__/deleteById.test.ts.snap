// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteBuilder pg basic queries no additional options 1`] = `"delete from \\"actor\\" where \\"actor\\".\\"actor_id\\" = ?"`;

exports[`DeleteBuilder pg basic queries no additional options 2`] = `
Array [
  20,
]
`;
